connmanager/outgoing_connections.go

```diff
@@ -3,40 +3,39 @@ package connmanager
 // checkOutgoingConnections goes over all activeOutgoing and makes sure they are still active.
 // Then it opens connections so that we have targetOutgoing active connections
 func (c *ConnectionManager) checkOutgoingConnections(connSet connectionSet) {
-	liveConnections := 0
 	for address := range c.activeOutgoing {
-		connection := connSet.get(address)
-		if connection != nil { // connections still connected
+		connection, ok := connSet.get(address)
+		if ok { // connections still connected
 			connSet.remove(connection)
-			liveConnections++
 			continue
 		}
 
 		// if connection is dead - remove from list of active ones
 		delete(c.activeOutgoing, address)
 	}
 
-	connectionsToAdd := c.targetOutgoing - liveConnections
-	if connectionsToAdd == 0 {
+	liveConnections := len(c.activeOutgoing)
```

Comment:
In the for loop we use `len(c.activeOutgoing)` and everywhere else we use `liveConnections`. I think for simplicity it'll be easier to always use `len(c.activeOutgoing)`

Reply:
I re-use liveConnections in 3 places, would be silly doing something like:
```
	log.Debugf("Have got %d outgoing connections out of target %d, adding %d more",
		len(c.activeOutgoing), c.targetOutgoing, c.targetOutgoing-len(c.activeOutgoing))
```

On the other hand, the loop requires an update value.
If you prefer, I can update the value of `liveConnections` in the end of the loop.